File: SofiForce.Models/Models/EntityModels/PromotionCriteriaModel.cs
using System;

namespace Models
{
    public class PromotionCriteriaModel
    {
        public int CriteriaId { get; set; }
        public int? PromotionId { get; set; }


        public int? GroupId { get; set; }
        public int? GroupNo { get; set; }
        public double? Slice { get; set; }

        public int? AttributeId { get; set; }
        public string AttributeCode { get; set; }

        public string ValueFrom { get; set; }
        public bool? IsActive { get; set; }
        public bool? Excluded { get; set; }
        public int? CBy { get; set; }
        public DateTime? CDate { get; set; }
        public int? EBy { get; set; }
        public DateTime? EDate { get; set; }
    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/PromotionCriteriaModel.ts
export interface PromotionCriteriaModel {
    criteriaId: number;

    promotionId: number | null;
    groupId: number | null;
    groupCode: string;

    attributeId: number | null;
    attributeCode: string | null;

    valueFrom: string;

    isActive: boolean | null;
    excluded: boolean | null;
    cBy: number | null;
    cDate: Date | null;
    eBy: number | null;
    eDate: Date | null;
}